### ISSUE TITLE
wp-env stalls during 'Starting WordPress' with PHP 8.1 or higher

### ISSUE THREAD

**Issue by braveryk7** [WordPress/gutenberg#51620]
### Description

Hi.

When using wp-env and specifying a phpVersion of PHP8.1 or later, the progress stops after 'Starting WordPress' is displayed in the console.
This issue occurs not only in a local environment but also in GitHub Actions CI.
Specifically, CI succeeds with PHP8.0 but terminates with an error when using PHP8.1 or PHP8.2.

This workflow was functioning without any issues until about a week ago. I can confirm that the workflow for the plugin I am developing was running successfully.

* [braveryk7/admin-bar-tools](https://github.com/braveryk7/admin-bar-tools/actions/runs/5251214979)

I'm not sure if it's related, but I came across the following tweet:

https://twitter.com/debian_jp/status/1668608983324827648?s=20

Although it's written in Japanese, the tweet states, "There is currently an issue where the 'Release file has expired' error occurs when using apt or apt-get commands, preventing Debian from being updated. If you encounter this issue, please modify /etc/apt/sources.list from http://ftp.jp.debian.org to http://deb.debian.org and try updating again."

It seems that there is an error occurring with Debian in Japan.

Interestingly, the problematic part also involves /etc/apt/sources.list. Could this be related? I am Japanese and residing in Japan. If GitHub Actions change regions based on user residence, it might be a specific issue within Japan.

### Step-by-step reproduction instructions

1. Specify PHP8.1 or PHP8.2 in the phpVersion of the .wp-env.json file.
2. Execute the command 'npm run wp-env start' in the console.
3. The process stops after the message 'Starting WordPress' is displayed.

### Screenshots, screen recording, code snippet

Below is the output of the docker ps -a command when the issue occurs, showing that the wordpress/tests-wordpress/cli/tests-cli containers are not running.

```
$ docker ps -a
CONTAINER ID   IMAGE     COMMAND                   CREATED         STATUS         PORTS                     NAMES
60d0b51287e5   mariadb   "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:54087->3306/tcp   002ffeb82c724fe8e97a0f4b2f773a95-mysql-1
```

Additionally, the log of the GitHub Actions workflow is provided.

```
Run npm run wp-env start

> wp-env
> wp-env start

✖ Error while running docker-compose command.
Sending build context to Docker daemon  456.6MB

Step 1/23 : FROM wordpress:php8.1
php8.1: Pulling from library/wordpress
5b5fe70539cd: Pulling fs layer
affe9439d2a2: Pulling fs layer
1684de57270e: Pulling fs layer
dc968f4da64f: Pulling fs layer
9b1cf1bf39c5: Pulling fs layer
719f04c53973: Pulling fs layer
74b24852707c: Pulling fs layer
54373d1fcd4e: Pulling fs layer
Building wordpress
The command '/bin/sh -c sed -i 's|deb.debian.org/debian stretch|archive.debian.org/debian stretch|g' /etc/apt/sources.list' returned a non-zero code: 2
Service 'wordpress' failed to build : Build failed
27a6a57081a1: Pulling fs layer
d40c0e518300: Pulling fs layer
36b134b9a8c3: Pulling fs layer
91324d697396: Pulling fs layer
9d78e346f9b5: Pulling fs layer
1e377d3ada5b: Pulling fs layer
74d16ae3ae54: Pulling fs layer
b0a7c5a4e6f2: Pulling fs layer
e311fd5817ca: Pulling fs layer
2a04720422ca: Pulling fs layer
56ddfa9ed2a2: Pulling fs layer
0a0c8627e7de: Pulling fs layer
f57035db186f: Pulling fs layer
dc968f4da64f: Waiting
9b1cf1bf39c5: Waiting
719f04c53973: Waiting
74b24852707c: Waiting
54373d1fcd4e: Waiting
27a6a57081a1: Waiting
d40c0e518300: Waiting
36b134b9a8c3: Waiting
91324d697396: Waiting
9d78e346f9b5: Waiting
1e377d3ada5b: Waiting
e311fd5817ca: Waiting
74d16ae3ae54: Waiting
b0a7c5a4e6f2: Waiting
0a0c8627e7de: Waiting
2a04720422ca: Waiting
56ddfa9ed2a2: Waiting
f57035db186f: Waiting
affe9439d2a2: Verifying Checksum
affe9439d2a2: Download complete
5b5fe70539cd: Verifying Checksum
5b5fe70539cd: Download complete
dc968f4da64f: Verifying Checksum
dc968f4da64f: Download complete
719f04c53973: Verifying Checksum
719f04c53973: Download complete
9b1cf1bf39c5: Verifying Checksum
9b1cf1bf39c5: Download complete
74b24852707c: Verifying Checksum
74b24852707c: Download complete
1684de57270e: Verifying Checksum
1684de57270e: Download complete
27a6a57081a1: Verifying Checksum
27a6a57081a1: Download complete
54373d1fcd4e: Verifying Checksum
54373d1fcd4e: Download complete
36b134b9a8c3: Verifying Checksum
36b134b9a8c3: Download complete
d40c0e518300: Verifying Checksum
d40c0e518300: Download complete
91324d697396: Verifying Checksum
91324d697396: Download complete
9d78e346f9b5: Verifying Checksum
9d78e346f9b5: Download complete
74d16ae3ae54: Verifying Checksum
74d16ae3ae54: Download complete
b0a7c5a4e6f2: Verifying Checksum
b0a7c5a4e6f2: Download complete
1e377d3ada5b: Verifying Checksum
1e377d3ada5b: Download complete
e311fd5817ca: Verifying Checksum
e311fd5817ca: Download complete
2a04720422ca: Verifying Checksum
2a04720422ca: Download complete
0a0c8627e7de: Verifying Checksum
0a0c8627e7de: Download complete
f57035db186f: Verifying Checksum
f57035db186f: Download complete
56ddfa9ed2a2: Verifying Checksum
56ddfa9ed2a2: Download complete
5b5fe70539cd: Pull complete
affe9439d2a2: Pull complete
1684de57270e: Pull complete
dc968f4da64f: Pull complete
9b1cf1bf39c5: Pull complete
719f04c53973: Pull complete
74b24852707c: Pull complete
54373d1fcd4e: Pull complete
27a6a57081a1: Pull complete
d40c0e518300: Pull complete
36b134b9a8c3: Pull complete
91324d697396: Pull complete
9d78e346f9b5: Pull complete
1e377d3ada5b: Pull complete
74d16ae3ae54: Pull complete
b0a7c5a4e6f2: Pull complete
e311fd5817ca: Pull complete
2a04720422ca: Pull complete
56ddfa9ed2a2: Pull complete
0a0c8627e7de: Pull complete
f57035db186f: Pull complete
Digest: sha256:63f850a5920b61983e513f8e7966c1aa4ee47b10cc5adfdb52670f3bd1ea620e
Status: Downloaded newer image for wordpress:php8.1
 ---> 9c2c966db4aa
Step 2/23 : RUN sed -i 's|deb.debian.org/debian stretch|archive.debian.org/debian stretch|g' /etc/apt/sources.list
 ---> Running in da06c4c220c1
sed: can't read /etc/apt/sources.list: No such file or directory

Error: Process completed with exit code 1.
```

### Environment info

```
$ docker -v
Docker version 24.0.2, build cb74dfc

$ docker-compose -v
Docker Compose version v2.18.1

$ node -v
v18.15.0

$ npm -v
9.5.0
```

@wordpress/env npm package -> v8.1.0 / 8.0.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by braveryk7** [comment]
I apologize for the oversight.

Currently, this issue is occurring on macOS in my local environment.

Also, there might be a related issue in the docker-library as well. You can find it here: https://github.com/docker-library/wordpress/issues/833.

**Comment by t-hamano** [comment]
Hi @braveryk7,

The issue you reported appears to be similar to the issue fixed in #51513. Version 8.1.1 of the env package was recently released with this fix, so could you please check if the problem reproduces with this version as well?

**Comment by braveryk7** [comment]
Hi @t-hamano

Thank you for pointing me to the relevant pull request. I had checked the issue but hadn't managed to look into the pull requests. Following your suggestion, I tried out version 8.1.1 of the env package and it worked perfectly. 

I want to express my sincere gratitude for your support. It was instrumental in resolving this issue. 

With this response, I am going to close this issue as it has been successfully resolved. 

Thank you again for your help.

Best regards,
